logstash-core/src/main/java/org/logstash/common/io/RecordIOReader.java

```diff
@@ -55,8 +59,12 @@ public RecordIOReader(Path path) throws IOException {
         ByteBuffer versionBuffer = ByteBuffer.allocate(1);
         this.channel.read(versionBuffer);
         versionBuffer.rewind();
-        if (versionBuffer.get() != VERSION) {
-            throw new RuntimeException("Invalid file. check version");
+        byte versionInFile = versionBuffer.get();
+        if (versionInFile != VERSION) {
+            logger.debug(String.format("File path: %s", path));
```

Comment:
Instead of a separate debug log entry, could you include the path to the file in the exception's message? Perhaps something like:

`"Invalid version on PQ data file %s. Expected: %c. Found: %c"`

Reply:
good idea. done.